### PR TITLE
add a test for jobs with setPreference: false

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR 
supersedes #12209
**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

no cl no fun

### CODE REVIEW

File: Content.IntegrationTests/Tests/Station/StationJobsTest.cs [hunk 1]
```diff
@@ -188,6 +189,40 @@ public async Task AdjustJobsTest()
...
+
+            foreach (var gameMap in prototypeManager.EnumeratePrototypes<GameMapPrototype>())
+            {
+                foreach (var (stationId, station) in gameMap.Stations)
+                {
+                    foreach (var job in station.AvailableJobs.Keys)
+                    {
+                        Assert.That(invalidJobs.Contains(job), Is.False, $"Staion {stationId} contains job prototype {job} which cannot be present roundstart.");
+                    }
+                }
```
Comment (moonheart08) [inline Content.IntegrationTests/Tests/Station/StationJobsTest.cs:220]: ....what.
this just takes the map and then does nothing with it

**Review by DrSmugleaf — APPROVED** [review]

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Did you mean to delete dart

**Comment by EmoGarbage404** [comment]
> Did you mean to delete dart

moony told me to

**Comment by AJCM-git** [comment]
merge master and see if the test still works

**Pull request merged**